platforms/software/dependency-management/src/main/java/org/gradle/internal/component/model/GraphSelectionCandidates.java

```diff
@@ -37,5 +39,8 @@ public interface GraphSelectionCandidates {
      * Returns the variant that is identified by the given configuration name.
      */
     @Nullable
-    VariantGraphResolveState getVariantByConfigurationName(String name);
+    default VariantGraphResolveState getVariantByConfigurationName(String name)  {
```

Comment:
❗ This is overriden by a few types - shouldn't these method overrides be removed so that subtypes also throw this exception?

Alternatively, it would be good to either deprecate + update Javadoc on this method or just remove it entirely to avoid the internal API looking like it makes this functionality available.

Reply:
Switched to instanceof + casting instead of having an interface method that only some implementations implement